Mage.Server/src/test/java/mage/server/util/ConfigWrapperTest.java

```diff
@@ -0,0 +1,294 @@
+    }
+
+
+    private final Comparator<Plugin> pluginComparator = (p1, p2) -> {
+        if (Objects.equals(p1.getName(), p2.getName()) &&
+                Objects.equals(p1.getJar(), p2.getJar()) &&
+                Objects.equals(p1.getClassName(), p2.getClassName())) {
+            return 0;
+        } else {
+            return -1;
+        }
+    };
+
+    private final Comparator<GamePlugin> gamePluginComparator = (p1, p2) -> {
+        if (Objects.equals(p1.getName(), p2.getName()) &&
+                Objects.equals(p1.getJar(), p2.getJar()) &&
+                Objects.equals(p1.getClassName(), p2.getClassName()) &&
+                Objects.equals(p1.getTypeName(), p2.getTypeName())) {
+            return 0;
+        } else {
+            return -1;
+        }
+    };
+
+    private final List<Plugin> randomPlugins = IntStream.range(0, RandomUtils.nextInt(1, 10))
+            .mapToObj(i -> makePlugin(
+                    RandomStringUtils.randomAlphanumeric(15),
+                    RandomStringUtils.randomAlphanumeric(16),
+                    RandomStringUtils.randomAlphanumeric(17))
+            ).collect(Collectors.toList());
+    private final List<GamePlugin> randomGamePlugins = IntStream.range(0, RandomUtils.nextInt(1, 10))
+            .mapToObj(i -> makeGamePlugin(
+                    RandomStringUtils.randomAlphanumeric(15),
+                    RandomStringUtils.randomAlphanumeric(16),
+                    RandomStringUtils.randomAlphanumeric(17),
+                    RandomStringUtils.randomAlphanumeric(18))
+            ).collect(Collectors.toList());
+
+    private Plugin makePlugin(String name, String jar, String className) {
+        final Plugin plugin = new Plugin();
+        plugin.setName(name);
+        plugin.setJar(jar);
+        plugin.setClassName(className);
+        return plugin;
+    }
+
+    private GamePlugin makeGamePlugin(String name, String jar, String className, String typeName) {
+        final GamePlugin plugin = new GamePlugin();
+        plugin.setName(name);
+        plugin.setJar(jar);
+        plugin.setClassName(className);
+        plugin.setTypeName(typeName);
+        return plugin;
+    }
+
+    @TestFactory
+    @DisplayName("should extract")
+    Stream<DynamicTest> pluginsExtraction() {
+        return Stream.of(
+                pluginTest("playerTypes from playerTypes", c -> c.playerTypes = randomPlugins, ConfigWrapper::getPlayerTypes),
+                gamePluginTest("gameTypes from gameTypes", c -> c.gameTypes = randomGamePlugins, ConfigWrapper::getGameTypes),
+                gamePluginTest("tournamentTypes from tournamentTypes", c -> c.tournamentTypes = randomGamePlugins, ConfigWrapper::getTournamentTypes),
+                pluginTest("draftCubes from draftCubes", c -> c.draftCubes = randomPlugins, ConfigWrapper::getDraftCubes),
+                pluginTest("deckTypes from deckTypes", c -> c.deckTypes = randomPlugins, ConfigWrapper::getDeckTypes)
```

Comment:
These tests verify that the collection in the configuration are loaded as expected.

Reply:
1. I see a random values tests. What about real config test?

Reply:
If you mean, values read from a config file, they are verified in `ConfigFactoryTest`. Since the wrapper is just using a POJO `Config` and performing deep access in certain fields, using random values guarantees that the test will fail at some point if there is any special logic implemented. So, using random values provides stronger verification than using constant values.